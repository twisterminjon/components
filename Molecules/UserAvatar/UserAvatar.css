.useravatar-container {
  position: relative;
}

.useravatar-statusbubble {
  position: absolute;
  right: 0;
  bottom: 0;
}
