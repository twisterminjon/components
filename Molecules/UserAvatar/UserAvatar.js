import React from 'react'
import PropTypes from 'prop-types'
import Avatar from '../../Atoms/Avatar/Avatar'
import StatusBubble from '../../Atoms/StatusBubble/StatusBubble'
import { USER_STATUS_LIST, USER_STATUS_BUSY, USER_STATUS_OFFLINE } from '../../../constants'
import './UserAvatar.css'

export default function UserAvatar({ size, profileImage, status, ...rest }) {
  return (
    <div className="useravatar-container" {...rest}>
      <Avatar imgUrl={profileImage} size={size} dimmed={[USER_STATUS_BUSY, USER_STATUS_OFFLINE].includes(status)} />
      <StatusBubble className="useravatar-statusbubble" status={status} />
    </div>
  )
}

// FIXME: This should not take a 'number' as a size. It should always be one of the strings.
UserAvatar.propTypes = {
  /** Size for the avatar */
  size: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.oneOf(['small', 'medium', 'large', 'big', 'huge', 'massive']),
  ]),
  /** Url to an image to display */
  profileImage: PropTypes.string,
  /** The status to to display in the bubble */
  status: PropTypes.oneOf(USER_STATUS_LIST).isRequired,
}

UserAvatar.defaultProps = {
  size: 32,
  profileImage: '',
}
