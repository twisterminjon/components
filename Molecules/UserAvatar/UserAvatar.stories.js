import React from 'react'

import { storiesOf } from '@storybook/react'
import centered from '@storybook/addon-centered/react'
import { withKnobs, boolean, number, select } from '@storybook/addon-knobs'
import UserAvatar from './UserAvatar'
import { USER_STATUS_AVAILABLE, USER_STATUS_BUSY, USER_STATUS_AWAY, USER_STATUS_OFFLINE } from '../../../constants'

const stories = storiesOf('Provider/v1/Molecules', module)
stories.addDecorator(withKnobs)
stories.addDecorator(centered)

stories.add('UserAvatar', () => {
  const size = number('size', 32)

  const toShowProfileImage = boolean('profileImage', false)
  const profileImage = toShowProfileImage
    ? 'https://tinyfac.es/data/avatars/852EC6E1-347C-4187-9D42-DF264CCF17BF-200w.jpeg'
    : ''

  const statusOptions = {
    available: USER_STATUS_AVAILABLE,
    busy: USER_STATUS_BUSY,
    away: USER_STATUS_AWAY,
    offline: USER_STATUS_OFFLINE,
  }
  const status = select('status', statusOptions, USER_STATUS_AVAILABLE)

  return <UserAvatar size={size} profileImage={profileImage} status={status} />
})
