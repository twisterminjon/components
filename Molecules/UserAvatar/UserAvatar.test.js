import React from 'react'
import UserAvatar from './UserAvatar'

describe('UserAvatar', () => {
  const size = 24
  const profileImage = 'https://www.fillmurray.com/60/60'

  it('matches the snapshot', () => {
    const wrapper = window.shallow(<UserAvatar size={size} profileImage={profileImage} status="available" />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders with an image', () => {
    const wrapper = window.shallow(<UserAvatar size={size} profileImage={profileImage} status="available" />)
    expect(wrapper.find('img')).toBeTruthy()
  })

  it('shows placeholder when no image is provided', () => {
    const wrapper = window.mount(<UserAvatar size={size} status="available" />)
    expect(wrapper.containsMatchingElement(<img alt="user avatar" src="user-placeholder.png" />)).toBeTruthy()
  })
})
