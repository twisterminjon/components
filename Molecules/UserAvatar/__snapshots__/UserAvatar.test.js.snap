// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAvatar matches the snapshot 1`] = `
<div
  className="useravatar-container"
>
  <Avatar
    className=""
    dimmed={false}
    displayName={null}
    imgUrl="https://www.fillmurray.com/60/60"
    loading={false}
    size={24}
    style={Object {}}
  />
  <StatusBubble
    className="useravatar-statusbubble"
    status="available"
    style={Object {}}
  />
</div>
`;
